### DISCUSSION TITLE
Ability to filter only access representations within bitstream in client.bitstreams_for_asset(asset) for PAX objects

### DISCUSSION THREAD

**Discussion by vchettur** [carj/pyPreservica#20] [General]
Hi James,
I have a PAX object that was produced as a sample for a digitization project. It has 1050 tiffs and a single PDF as an access copy which is about 500 MBs. I can use the Content API to download to my desktop in about 15-20 seconds. However, the issue with the content API is that if there are jpegs corresponding to the tiffs instead of a single pdf, it won't work. On the other hand, I can use the Entity API and eventually download it. 

Using the shortcut method: client.bitstreams_for_asset(asset) and filtering on .pdf also works but it appears that we have to loop through every item to get to the .pdf

I'm not sure if there is a better way to access the Access Representation quickly.

**Comment by vchettur** [comment]
I think I have found what I need to do by using the low level Preservica Entity API endpoints and chunked downloads.

**Comment by carj** [comment]
The following should give you just the access representation files


```python
for representation in client.representations(asset):
    if representation.rep_type  == "Access":
        for content_object in client.content_objects(representation):
            for generation in client.generations(content_object):
                for bitstream in generation.bitstreams:
                    client.bitstream_content(bitstream, bitstream.filename)
```

**Reply by vchettur** [reply]
Thanks, James. I had the filter, representation.rep_type  == "Access", too low in the loop in the test using the long method. It performs better now. In production, I have to have the filter at a lower level since sometimes the Access representation is also a Preservation representation, e.g. sometimes we receive only pdfs, jpgs, mp3 etc. I think I can handle all the use cases by tweaking the filters appropriately.
